binding_generator.py

```diff
@@ -907,6 +912,47 @@ def generate_builtin_class_header(builtin_api, size, used_classes, fully_used_cl
         result.append("\tVariant &operator[](int64_t p_index);")
         result.append("\tvoid set_typed(uint32_t p_type, const StringName &p_class_name, const Variant &p_script);")
         result.append("\tvoid _ref(const Array &p_from) const;")
+        result.append("""
+	struct Iterator {
+		_FORCE_INLINE_ Variant &operator*() const;
+		_FORCE_INLINE_ Variant *operator->() const;
+		_FORCE_INLINE_ Iterator &operator++();
+		_FORCE_INLINE_ Iterator &operator--();
```

Comment:
I think the body of these functions should be inlined here, rather than put into `variant.hpp`, just like we're doing with the packed arrays a few lines above

Reply:
They can't be inlined without including `variant.hpp` since they need to know the size of Variant to increment or decrement the pointer, thats why I put them in `variant.hpp` and I assume why they're in the in the godot repo too.